pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableImplV2.java

```diff
@@ -188,10 +193,13 @@ public void addException(ProcessingException processingException) {
     }
 
     // Write metadata.
-    dataOutputStream.writeInt(dataOffset);
-    byte[] metadataBytes = serializeMetadata();
-    dataOutputStream.writeInt(metadataBytes.length);
-    dataOffset += metadataBytes.length;
+    byte[] metadataBytes = null;
```

Comment:
This won't work as the deserializer expect the metadata to be present even if it is empty

Reply:
good catch! made it write 0 length metadataBytes